File: web/src/layout/DateRangeDemoXml.ts
//
// DateRangeDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly-web/dist/views/html'
import { customViewSetDelegate } from 'butterfly-web/dist/views/CustomView'
import { MonthCVD } from 'butterfly-web/dist/views/MonthCVD'
import { SelectDateMonthCVD } from 'butterfly-web/dist/views/SelectDateMonthCVD'
import { SelectDateRangeMonthCVD } from 'butterfly-web/dist/views/SelectDateRangeMonthCVD'
import { SelectMultipleDatesMonthCVD } from 'butterfly-web/dist/views/SelectMultipleDatesMonthCVD'
import htmlForDefault from './date_range_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.DateRangeDemoXml
export class DateRangeDemoXml {
    xmlRoot!: HTMLElement
    month!: HTMLCanvasElement
    month2!: HTMLCanvasElement
    month3!: HTMLCanvasElement
    month4!: HTMLCanvasElement
    monthDelegate!: MonthCVD
    month2Delegate!: SelectDateMonthCVD
    month3Delegate!: SelectDateRangeMonthCVD
    month4Delegate!: SelectMultipleDatesMonthCVD
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.month = getViewById<HTMLCanvasElement>(view, "month")
        this.month2 = getViewById<HTMLCanvasElement>(view, "month2")
        this.month3 = getViewById<HTMLCanvasElement>(view, "month3")
        this.month4 = getViewById<HTMLCanvasElement>(view, "month4")
        if(this.month){ this.monthDelegate = new MonthCVD(); customViewSetDelegate(this.month, this.monthDelegate); }
        if(this.month2){ this.month2Delegate = new SelectDateMonthCVD(); customViewSetDelegate(this.month2, this.month2Delegate); }
        if(this.month3){ this.month3Delegate = new SelectDateRangeMonthCVD(); customViewSetDelegate(this.month3, this.month3Delegate); }
        if(this.month4){ this.month4Delegate = new SelectMultipleDatesMonthCVD(); customViewSetDelegate(this.month4, this.month4Delegate); }
        return view
    }
}

File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/DateRangeDemoXml.kt
//
// DateRangeDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class DateRangeDemoXml {

    lateinit var month: com.lightningkite.butterfly.views.widget.CustomView
    lateinit var month2: com.lightningkite.butterfly.views.widget.CustomView
    lateinit var month3: com.lightningkite.butterfly.views.widget.CustomView
    lateinit var month4: com.lightningkite.butterfly.views.widget.CustomView
    lateinit var monthDelegate: com.lightningkite.butterfly.views.MonthCVD
    lateinit var month2Delegate: com.lightningkite.butterfly.views.SelectDateMonthCVD
    lateinit var month3Delegate: com.lightningkite.butterfly.views.SelectDateRangeMonthCVD
    lateinit var month4Delegate: com.lightningkite.butterfly.views.SelectMultipleDatesMonthCVD
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.date_range_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        month = view.findViewById<com.lightningkite.butterfly.views.widget.CustomView>(R.id.month)
        month2 = view.findViewById<com.lightningkite.butterfly.views.widget.CustomView>(R.id.month2)
        month3 = view.findViewById<com.lightningkite.butterfly.views.widget.CustomView>(R.id.month3)
        month4 = view.findViewById<com.lightningkite.butterfly.views.widget.CustomView>(R.id.month4)
        monthDelegate = view.findViewById<CustomView>(R.id.month).delegate as com.lightningkite.butterfly.views.MonthCVD
        month2Delegate = view.findViewById<CustomView>(R.id.month2).delegate as com.lightningkite.butterfly.views.SelectDateMonthCVD
        month3Delegate = view.findViewById<CustomView>(R.id.month3).delegate as com.lightningkite.butterfly.views.SelectDateRangeMonthCVD
        month4Delegate = view.findViewById<CustomView>(R.id.month4).delegate as com.lightningkite.butterfly.views.SelectMultipleDatesMonthCVD
        
        
        return view
    }
}